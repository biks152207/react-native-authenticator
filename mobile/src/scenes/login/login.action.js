import { AUTH } from '../../app/common/enums';
import { asyncStorage, authErrorBuilder, loginWithFacebook, processFormCallback} from '../../app/common/helper';
import { asyncActionNames, buildAsyncActions} from '../services/actionCreator';
import { Actions } from 'react-native-router-flux';


// Build action names for login
const actionNames = asyncActionNames('LOGIN');
const actionCreators = buildAsyncActions(actionNames);

export function processForm({email, password}) {
	return function(dispatch) {
		dispatch(actionCreators.progress())
		fetch(AUTH.LOGIN, {
			method: 'POST',
			headers: {
		    'Accept': 'application/json',
		    'Content-Type': 'application/json',
		  },
			body: JSON.stringify({
		    email: email,
		    password: password,
		  })
		})
		.then((response) => response.json())
		.then((responseData) => {
				processFormCallback(responseData, dispatch);
		})
		.catch((error) => {
				authErrorBuilder(error)
					.then((res) =>{
						dispatch(actionCreators.failure(res));
					})
		}).done();
	};
};

export  function facebookLogin(){
	return function(dispatch){
		dispatch(actionCreators.progress());
		loginWithFacebook()
			.then((result) =>{
				let user = {
          name: result.user.name,
          email: result.user.email,
          social: {
            facebook: {
              id: result.user.id,
              token: result.accessToken
            }
          }
        }
				fetch(AUTH.FACEBOOK, {
					method: 'POST',
					headers: {
				    'Accept': 'application/json',
				    'Content-Type': 'application/json',
				  },
					body: JSON.stringify(user)
				})
				.then((response) => response.json())
				.then((responseData) =>{
					processFormCallback(responseData, dispatch);
				})
			})
	}
}
