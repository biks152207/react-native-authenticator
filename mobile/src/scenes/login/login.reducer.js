
import { asyncActionNames } from '../services/actionCreator';
const actionNames = asyncActionNames('LOGIN');

const INITIAL_STATE = { errors: null, authenticated: false, admin_privileges: false, profile: null, progress: false};

export default function (state = INITIAL_STATE, action) {
	switch(action.type) {
       case actionNames.success:
        return { ...state, errors: null, authenticated: true, profile: action.payload, progress: false };
      case actionNames.failure:
        return { ...state, errors: action.errors, progress: false };
      case actionNames.unauth:
        return { ...state, errors: null, authenticated: false, profile: null, progress: false}
      case actionNames.progress:
        return { ...state, progress: true}
      case actionNames.privileges:
        return { ...state, admin_privileges: true };
	}
	return state;
};
