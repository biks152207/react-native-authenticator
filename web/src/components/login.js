import React, { Component } from 'react';
import { Link, browserHistory } from 'react-router';
import { Card, CardTitle, CardText, RaisedButton, TextField } from 'material-ui';
import axios from 'axios';
import { connect } from 'react-redux';

class Login extends Component {

	/**
   * Class constructor.
   */
  constructor() {
    super();

    // set the initial component state
    this.state = {
      errorMessage: '',
      errors: {},
      authenticated: false,
      admin_privileges: false
    };
    this.processForm = this.processForm.bind(this)
  }

	processForm(event) {
		event.preventDefault();
			let self = this;

			let state = {};

			axios('http://localhost:3000/auth/login', {
				method: 'POST',
				data: { 
					email: this.refs.email.getValue(),
		            password: this.refs.password.getValue()
		        }
			}).then((response) => {

			    if (response.status == 200) {
			        // success

			        this.setState({
					      errorMessage: '',
					      errors: {},
					      authenticated: true,
					      admin_privileges: true
					});
			       
			        localStorage.setItem('token', response.data.token);
			       	localStorage.setItem('user', JSON.stringify(response.data.userData));

			        browserHistory.push('/home');
			        
			    }
			}).catch((error) => {
	        	// change the component state
	        	this.setState({
	        		errorMessage: error.response.data.message,
	        		errors: error ? error.response.data.errors : {}
	        	});
			});
		
	}

	render() {
		return (
			<Card class="container">
				<form action="/" onSubmit={this.processForm}>
					<h2 className="card-heading">Log In</h2>

					<CardTitle title="Login with Email" />

					{this.state.errorMessage && <p className="error-message">{this.state.errorMessage}</p>}

					<div className="field-line">
						<TextField ref="email" floatingLabelText="Email" 
						errorText={this.state.errors.email}/>
					</div>

					<div className="field-line">
						<TextField ref="password" floatingLabelText="Password"
						errorText={this.state.errors.password} />
					</div>

					<div className="button-line">
						<RaisedButton type="submit" label="Login" primary={true} />
					</div>

					<CardText>Dont have an account?<Link to={`/signup`}>Create One</Link></CardText>
				</form>
			</Card>
		);
	}

	/*function mapStateToProps(state) {
    	return { authenticated: state.auth.authenticated, admin_privileges: state.auth.admin_privileges };
  	}

  	return connect(mapStateToProps)(Login);*/
}

export default Login;