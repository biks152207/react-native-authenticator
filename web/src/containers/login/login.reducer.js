import { 
  AUTH_USER,
  SET_ADMIN_PRIVILEGES,
  AUTH_ERROR
} from './login.types';

const INITIAL_STATE = { errors: null, authenticated: false, admin_privileges: false };

export default function (state = INITIAL_STATE, action) {
	switch(action.type) {
		 case AUTH_USER:
	      return { ...state, errors: null, authenticated: true };
	     case SET_ADMIN_PRIVILEGES:
	      return { ...state, admin_privileges: true }; 
	     case AUTH_ERROR:
	      return { ...state, errors: action.errors }; 
	}
	return state;
};