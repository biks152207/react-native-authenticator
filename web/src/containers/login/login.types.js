export const AUTH_USER = 'auth_user';
export const SET_ADMIN_PRIVILEGES = 'set_admin_privileges';
export const AUTH_ERROR = 'auth_error';