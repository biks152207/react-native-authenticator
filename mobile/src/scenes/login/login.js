import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { View, Text, Button, TextInput, AsyncStorage, StyleSheet, ScrollView, TouchableHighlight } from 'react-native';
import t from 'tcomb-form-native';
import { Actions } from 'react-native-router-flux';


// Custom modules
import { emailValidator } from '../../app/common/validations';
import { processSignupForm } from '../signup/signup.action';
import { processForm, facebookLogin } from './login.action';
import { ProgressIndicator } from '../common-scenes/progress';

// Stylesheet
import { Styles } from '../../app/common/styles';

const Form = t.form.Form;
let Email = t.refinement(t.String, emailValidator);
Email.getValidationErrorMessage = function (value, path, context) {
  return 'Invalid email address';
};
const formModel = t.struct({
  email: Email,
  password: t.String
});

let options = {
  fields: {
    password: {
      error: 'Password is required',
      password: true,
      secureTextEntry: true,
      autoCapitalize: 'none'
    },
    email: {
      autoCapitalize: 'none'
    }
  }
};

export class Login extends Component {

	constructor() {
	    super();
			this.handleFormSubmit = this.handleFormSubmit.bind(this);
      this.reconfigure = this.reconfigure.bind(this);
	};

	handleFormSubmit() {
    let value = this.refs.form.getValue();
    if (value) { // if validation fails, value will be null
      this.props.processForm(value);
    }
	};

  reconfigure(){
    AsyncStorage.removeItem('config');
    Actions.setup();
  }

	render() {
		return (
			<ScrollView keyboardShouldPersistTaps={'handled'}>
				<View style={Styles.container}>
            <Form
            ref="form"
            type={formModel}
            options={options}
          />
          <TouchableHighlight style={Styles.button} onPress={this.handleFormSubmit} underlayColor='#99d9f4'>
            <Text style={Styles.buttonText}>Login</Text>
          </TouchableHighlight>
            <Button color="#3B5998" title="Login with facebook" onPress={this.props.facebookLogin} />
          <Text onPress={Actions.signup} style={{textAlign: 'center'}}>Dont have an account?</Text>
          {__DEV__ && <TouchableHighlight style={Styles.button} onPress={this.reconfigure} underlayColor='#99d9f5'>
            <Text style={Styles.buttonText}>Re-configure</Text>
          </TouchableHighlight> }
				</View>
        <View style={Styles.progress} >
        {
          this.props.progress ? (<ProgressIndicator style={{justifyContent: 'center',alignItems: 'center'}} />):(<Text></Text>)
        }
        </View>
     </ScrollView>
		);
	};
};


function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    processForm: processForm,
    processSignupForm: processSignupForm,
    facebookLogin: facebookLogin
	}, dispatch);
};

function mapStateToProps(state) {
	return {
		errors: state.login.errors,
    progress: state.login.progress
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
