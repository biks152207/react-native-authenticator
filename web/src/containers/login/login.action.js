import { AUTH_USER, SET_ADMIN_PRIVILEGES, AUTH_ERROR} from './login.types';

export function signInAction() {
	return {
		type: AUTH_USER
	};
};

export function setAdminPrevilegeAction() {
	return {
		type: SET_ADMIN_PRIVILEGES
	};
};

export function signInErrorAction(errors) {
	return {
		type: AUTH_ERROR,
		errors
	};
};