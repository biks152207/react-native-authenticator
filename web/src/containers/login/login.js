import React, { Component } from 'react';
import { Link, browserHistory } from 'react-router';
import { Card, CardTitle, CardText, RaisedButton, TextField, FloatingActionButton } from 'material-ui';
import axios from 'axios';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { signInAction, setAdminPrevilegeAction, signInErrorAction } from './login.action';


class AllFields extends Component {
    constructor(props){
        super();
        /*
        console.log("Fields is ");
        console.dir(props);
        */
        this.state = {fields: props.fields};
    }

    render(){
        console.dir(this.state.fields);
        return <div>{this.state.fields.map((field, index) =>
                        <TextField key={field.key} value={field.name} floatingLabelText={field.label}
                            onChange={event => this.props.parent.saveData(index, event.target.value)}
                        />)}
              </div>
    }
}

export class Login extends Component {

	constructor() {
	    super();
	    this.state = {
            emailName: 'Email',
            passwordName: 'Password',
            edit: false,
	    	email: '',
	    	password: '',
            lastField: 3,
            fields: [{name: '1', label: 'Number', key: 1},
                     {name: 'hi', label: 'Good', key: 2}]
	    };
	    this.processForm = this.processForm.bind(this);
	};

	processForm(event) {
		event.preventDefault();
		
		axios('http://localhost:3000/auth/login', {
			method: 'POST',
			data: { 
				email: this.state.email,
	            password: this.state.password
	        }
		}).then((response) => {

		    if (response.status == 200) {
		        // success
				this.props.signIn();
				this.props.adminPrevilege();
		       
		        localStorage.setItem('token', response.data.token);
		       	localStorage.setItem('user', JSON.stringify(response.data.userData));

		        browserHistory.push('/home');
		        
		    }
		}).catch((errors) => {
        	// change the component state
        	this.props.setErrors(errors);
		});
	}

    doEdit(event){
        event.preventDefault();
        var old_edit = this.state.edit;
        console.log("do edit " + old_edit);
        this.setState({edit: !old_edit});
        // this.refs.edit.innerText = old_edit ? 'Stop Editing' : 'Edit';
        console.dir(event.target);
        // event.target.innerText = "Stop Editing";
    }

    getEdit(){
        if (this.state.edit){
            return "Stop Editing";
        }
        return "Edit";
    }

    saveData(field, value){
        var fields = this.state.fields;
        fields[field].name = value;
        this.setState({fields: fields});
    }

    addItem(){
        let newField = {'name': 'x', label: 'Field', key: this.state.lastField};
        let fields = this.state.fields;
        fields.push(newField);
        this.setState({fields: fields, lastField: this.state.lastField + 1});
    }

	render() {
		return (
			<Card class="container">
				<form action="/" onSubmit={this.processForm}>
					<h2 className="card-heading">Log In</h2>

					<CardTitle title="Login with Email" />

                    <button ref='edit' id='edit' type='button' onClick={this.doEdit.bind(this)}>{this.getEdit()}</button>

					<p className="error-message"></p>

                    {
                      function (){
                        if (this.state.edit){
                            return <div>
                                        <TextField value={this.state.emailName} floatingLabelText="Name"
                                           onChange={(event) => this.setState({emailName: event.target.value})} />
                                        <TextField value={this.state.passwordName} floatingLabelText="Name"
                                           onChange={(event) => this.setState({passwordName: event.target.value})} />
                                        <FloatingActionButton id='add' type='button' onClick={this.addItem.bind(this)}>+</FloatingActionButton>
                                   </div>

                        }
                      }.bind(this)()
                    }

					<div className="field-line">
						<TextField value={this.state.email} floatingLabelText={this.state.emailName}
						errorText={ (this.props.errors) ? this.props.errors : '' }
						onChange={(event) => this.setState({ email: event.target.value})} />
					</div>

					<div className="field-line">
						<TextField value={this.state.password} floatingLabelText={this.state.passwordName}
						errorText={ (this.props.errors) ? this.props.errors : '' } 
						onChange={(event) => this.setState({ password: event.target.value})}/>
					</div>

                    <AllFields parent={this} fields={this.state.fields} />

					<div className="button-line">
						<RaisedButton type="submit" label="Login" primary={true} />
					</div>
					<a href="http://localhost:3000/auth/facebook">Login with Facebook</a>	
					<CardText>Dont have an account?<Link to={`/signup`}>Create One</Link></CardText>
				</form>
			</Card>
		);
	};
};


function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    signIn: signInAction,
    adminPrevilege: setAdminPrevilegeAction,
    setErrors: signInErrorAction
  }, dispatch);
};

function mapStateToProps(state) {
	return {
		errors: state.login.errors
	};
};


export default connect(mapStateToProps, mapDispatchToProps)(Login);
