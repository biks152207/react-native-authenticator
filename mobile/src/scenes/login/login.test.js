import 'react-native';
import React from 'react';
import { Login } from './login';
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

it('renders correctly', () => {
  const component = shallow(
    <Login />
  );
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot();
});