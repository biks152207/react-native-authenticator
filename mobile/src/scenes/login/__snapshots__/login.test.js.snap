exports[`test renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    AAAA
  </Text>
</View>
`;
